lib/rbUtil.js

```diff
@@ -239,15 +239,19 @@ function HTTPError(response) {
     if (response.body && response.body.type) {
         this.message += ': ' + response.body.type;
     }
-
-    for (var key in response) {
-        this[key] = response[key];
-    }
+    Object.assign(this, response);
 }
 util.inherits(HTTPError, Error);
 
 rbUtil.HTTPError = HTTPError;
 
+function HTTPRedirect(response) {
```

Comment:
My concern is primarily about avoiding surprising, and making sure that we keep a sane layering of requests. Some of these requests might be just internal, and callers might not necessarily expect a redirect error to be thrown at them. It seems that this could lead to surprises when this happens while fetching the original HTML for a page, for example. Should the caller follow the redirect? Would the redirected URL make sense? How are redirects translated so that they make sense at the outer layer, taking into account possible rewrites?

I think we should think through these implications a bit more. I'm fine with doing a one-off change here if we can establish that surprises as described above won't happen.